astropy/units/tests/test_quantity_array_methods.py

```diff
@@ -73,6 +73,9 @@ def test_flat(self):
         assert q_flat[8] == -1. * u.km / u.s
         assert q[2, 2] == -1. * u.km / u.s
 
+        # also check q_flat copies properly
+        assert all(q_flat.copy() == q.flatten())
```

Comment:
Since the tests above are so much about ensuring view and copies, maybe here add an 
```
q_flat_copy = q_flat.copy()
assert np.all(q_flat_copy == q.flatten())
assert isinstance(q_flat_copy, u.Quantity)
assert not np.may_share_memory(q_flat_copy, q)
```